diff --git a/hwdb/20-usb-vendor-model.hwdb b/hwdb/20-usb-vendor-model.hwdb
index 3bb26ad..4a82769 100644
--- a/hwdb/20-usb-vendor-model.hwdb
+++ b/hwdb/20-usb-vendor-model.hwdb
@@ -56921,6 +56921,12 @@ usb:v2237*
 usb:v2237p4161*
  ID_MODEL_FROM_DATABASE=eReader White
 
+usb:v2252*
+ ID_VENDOR_FROM_DATABASE=HBGIC Technology Co., Ltd.
+
+usb:v2252p0106*
+ ID_MODEL_FROM_DATABASE=USB Keyboard Mouse
+
 usb:v225D*
  ID_VENDOR_FROM_DATABASE=Morpho
 
@@ -56954,6 +56960,12 @@ usb:v228D*
 usb:v228Dp0001*
  ID_MODEL_FROM_DATABASE=Terminal Bike Key Reader
 
+usb:v22A1*
+ ID_VENDOR_FROM_DATABASE=MHTF
+
+usb:v22A1p2801*
+ ID_MODEL_FROM_DATABASE=MHTF
+
 usb:v22A6*
  ID_VENDOR_FROM_DATABASE=Pie Digital, Inc.
 
diff --git a/hwdb/60-keyboard.hwdb b/hwdb/60-keyboard.hwdb
index 71aecd8..d5824cf 100644
--- a/hwdb/60-keyboard.hwdb
+++ b/hwdb/60-keyboard.hwdb
@@ -1378,6 +1378,68 @@ evdev:atkbd:dmi:bvn*:bvr*:bd*:svnVIA:pnK8N800:pvr*
 evdev:name:SIPODEV USB Composite Device:dmi:bvn*:bvr*:bd*:svnVIOS:pnLTH17:pvr*
  KEYBOARD_KEY_70073=f21                                 # Touchpad toggle
 
+###########################################################
+# WeTek
+###########################################################
+
+# WeTek Play OpenELEC Remote
+evdev:input:b0003v2252p0106*
+ KEYBOARD_KEY_c00e2=f6
+ KEYBOARD_KEY_c019e=mute
+ KEYBOARD_KEY_7001e=1
+ KEYBOARD_KEY_7001f=2
+ KEYBOARD_KEY_70020=3
+ KEYBOARD_KEY_70021=4
+ KEYBOARD_KEY_70022=5
+ KEYBOARD_KEY_70023=6
+ KEYBOARD_KEY_70024=7
+ KEYBOARD_KEY_70025=8
+ KEYBOARD_KEY_70026=9
+ KEYBOARD_KEY_7003a=esc
+ KEYBOARD_KEY_70027=0
+ KEYBOARD_KEY_7003b=i
+ KEYBOARD_KEY_7003d=c
+ KEYBOARD_KEY_7003c=homepage
+ KEYBOARD_KEY_70029=back
+ KEYBOARD_KEY_7003e=l
+ KEYBOARD_KEY_70040=f11
+ KEYBOARD_KEY_70050=left
+ KEYBOARD_KEY_70052=up
+ KEYBOARD_KEY_70051=down
+ KEYBOARD_KEY_7004f=right
+ KEYBOARD_KEY_70028=enter
+ KEYBOARD_KEY_c00e9=volumeup
+ KEYBOARD_KEY_c00ea=volumedown
+ KEYBOARD_KEY_c0204=f5
+ KEYBOARD_KEY_70044=v
+ KEYBOARD_KEY_7004b=pageup
+ KEYBOARD_KEY_7004e=pagedown
+ KEYBOARD_KEY_70041=f1
+ KEYBOARD_KEY_70042=f2
+ KEYBOARD_KEY_70043=f3
+ KEYBOARD_KEY_c01c2=f4
+ KEYBOARD_KEY_c00b6=previoussong
+ KEYBOARD_KEY_c00b4=rewind
+ KEYBOARD_KEY_c00b3=fastforward
+ KEYBOARD_KEY_c00b5=nextsong
+ KEYBOARD_KEY_c00b2=f7
+ KEYBOARD_KEY_c00cd=playpause
+ KEYBOARD_KEY_c00b7=stopcd
+
+# WeTek Core Remote
+evdev:input:b0003v22A1p2801*
+ KEYBOARD_KEY_70040=c
+ KEYBOARD_KEY_7003f=f5
+ KEYBOARD_KEY_7004a=homepage
+ KEYBOARD_KEY_70029=back
+ KEYBOARD_KEY_70050=left
+ KEYBOARD_KEY_70052=up
+ KEYBOARD_KEY_70051=down
+ KEYBOARD_KEY_7004f=right
+ KEYBOARD_KEY_70028=enter
+ KEYBOARD_KEY_c00e9=volumeup
+ KEYBOARD_KEY_c00ea=volumedown
+
 ###########################################################
 # Zepto
 ###########################################################
diff --git a/hwdb/usb.ids b/hwdb/usb.ids
index 0f21774..e1bcdee 100644
--- a/hwdb/usb.ids
+++ b/hwdb/usb.ids
@@ -18994,6 +18994,8 @@
 	6323  USB Electronic Scale
 2237  Kobo Inc.
 	4161  eReader White
+2252  HBGIC Technology Co., Ltd.
+	0106  USB Keyboard Mouse
 225d  Morpho
 	0001  FINGER VP Multimodal Biometric Sensor
 	0008  CBM-E3 Fingerprint Sensor
@@ -19005,6 +19007,8 @@
 	000e  MorphoAccess SIGMA Biometric Access Control Terminal
 228d  8D Technologies inc.
 	0001  Terminal Bike Key Reader
+22a1  MHTF
+	2801  MHTF
 22a6  Pie Digital, Inc.
 	ffff  PieKey "beta" 4GB model 4E4F41482E4F5247 (SM3251Q BB)
 22b8  Motorola PCS
