diff --git a/drivers/amlogic/rtc/aml_rtc.c b/drivers/amlogic/rtc/aml_rtc.c
index daeaf7a..f0b77db 100755
--- a/drivers/amlogic/rtc/aml_rtc.c
+++ b/drivers/amlogic/rtc/aml_rtc.c
@@ -121,7 +121,7 @@ int get_rtc_status(void)
 	static int rtc_fail = -1;
 	//if (rtc_fail < 0) {
 		if (check_osc_clk() < 0) {
-			printk("rtc clock error\n");
+//			printk("rtc clock error\n");
 			rtc_fail = 1;
 		} else
 			rtc_fail = 0;
