diff --git a/Makefile b/Makefile
index 60a6864..194ea50 100644
--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@
 # Copyright (C) 2013 Lutz Fiebach (lufie@openelec.tv)
 
 ADDON_NAME=service.libreelec.settings
-ADDON_VERSION=9.0.0
+ADDON_VERSION=9.0.1
 DISTRONAME:=LibreELEC
 
 SHELL=/bin/bash
diff --git a/src/defaults.py b/src/defaults.py
index f302540..58df7d1 100644
--- a/src/defaults.py
+++ b/src/defaults.py
@@ -80,7 +80,8 @@ system = {
 updates = {
     'ENABLED': not os.path.exists('/dev/.update_disabled'),
     'UPDATE_REQUEST_URL': 'https://update.libreelec.tv/updates.php',
-    'UPDATE_DOWNLOAD_URL': 'http://%s.libreelec.tv/%s',
+    'UPDATE_REQUEST_URL_ALTERNATIVE': 'http://libreelec.dtech.hu/updates.php',
+    'UPDATE_DOWNLOAD_URL': 'http://libreelec.dtech.hu/%s/%s',
     'LOCAL_UPDATE_DIR': '/storage/.update/',
 
     'RPI_FLASHING_TRIGGER': '/storage/.rpi_flash_firmware',
diff --git a/src/resources/lib/modules/updates.py b/src/resources/lib/modules/updates.py
index 5530fad..afce518 100644
--- a/src/resources/lib/modules/updates.py
+++ b/src/resources/lib/modules/updates.py
@@ -546,11 +546,23 @@ class updates:
                 self.oe.url_quote(version),
                 self.oe.url_quote(self.hardware_flags),
                 )
+            url_alt = '%s?i=%s&d=%s&pa=%s&v=%s&f=%s' % (
+                self.UPDATE_REQUEST_URL_ALTERNATIVE,
+                self.oe.url_quote(systemid),
+                self.oe.url_quote(self.oe.DISTRIBUTION),
+                self.oe.url_quote(self.oe.ARCHITECTURE),
+                self.oe.url_quote(version),
+                self.oe.url_quote(self.hardware_flags),
+                )
+
             if self.oe.BUILDER_NAME:
                url += '&b=%s' % self.oe.url_quote(self.oe.BUILDER_NAME)
+               url_alt += '&b=%s' % self.oe.url_quote(self.oe.BUILDER_NAME)
 
             self.oe.dbg_log('updates::check_updates_v2', 'URL: %s' % url, 0)
             update_json = self.oe.load_url(url)
+            self.oe.dbg_log('updates::check_updates_v2', 'URL: %s' % url_alt, 0)
+            update_json = self.oe.load_url(url_alt)
             self.oe.dbg_log('updates::check_updates_v2', 'RESULT: %s' % repr(update_json), 0)
             if update_json != '':
                 update_json = json.loads(update_json)
