diff --git a/rtl8723AU/os_dep/linux/sdio_intf.c b/rtl8723AU/os_dep/linux/sdio_intf.c
index e1c0166..eec8e10 100755
--- a/rtl8723AU/os_dep/linux/sdio_intf.c
+++ b/rtl8723AU/os_dep/linux/sdio_intf.c
@@ -126,6 +126,13 @@ static const struct sdio_device_id sdio_ids[] = {
 //	{ /* end: all zeroes */				},
 };
 
+#ifdef CONFIG_RTL8723A
+MODULE_ALIAS("sdio:c*v024Cd8723*");
+#endif
+#ifdef CONFIG_RTL8188E
+MODULE_ALIAS("sdio:c*v024Cd8179*");
+#endif
+
 static int rtw_drv_init(struct sdio_func *func, const struct sdio_device_id *id);
 static void rtw_dev_remove(struct sdio_func *func);
 static int rtw_sdio_resume(struct device *dev);
