diff --git a/drivers/amlogic/wifi/bcmdhd/dhd_config.c b/drivers/amlogic/wifi/bcmdhd/dhd_config.c
index a68af64..c0ab8e9 100644
--- a/drivers/amlogic/wifi/bcmdhd/dhd_config.c
+++ b/drivers/amlogic/wifi/bcmdhd/dhd_config.c
@@ -57,6 +57,7 @@ uint config_msg_level = CONFIG_ERROR_LEVEL;
 #define BCM43341B0_CHIP_REV     2
 #define BCM43241B4_CHIP_REV     5
 #define BCM4335A0_CHIP_REV      2
+#define BCM4335B0_CHIP_REV      1
 #define BCM4339A0_CHIP_REV      1
 #define BCM43455C0_CHIP_REV     6
 #define BCM4354A1_CHIP_REV      1
@@ -125,6 +126,13 @@ const static char *bcm4339a0_ag_fw_name[] = {
 	"fw_bcm4339a0_ag_mfg.bin"
 };
 
+const static char *bcm4335b0_ag_fw_name[] = {
+	"fw_bcm4335b0_ag.bin",
+	"fw_bcm4335b0_ag_apsta.bin",
+	"fw_bcm4335b0_ag_p2p.bin",
+	"fw_bcm4335b0_ag_mfg.bin"
+};
+
 const static char *bcm43455c0_ag_fw_name[] = {
 	"fw_bcm43455c0_ag.bin",
 	"fw_bcm43455c0_ag_apsta.bin",
@@ -485,6 +493,8 @@ dhd_conf_set_fw_name_by_chip(dhd_pub_t *dhd, char *fw_path)
 		case BCM4335_CHIP_ID:
 			if (chiprev == BCM4335A0_CHIP_REV)
 				strcpy(&fw_path[i+1], bcm4339a0_ag_fw_name[fw_type]);
+			else if (chiprev == BCM4335B0_CHIP_REV)
+				strcpy(&fw_path[i+1], bcm4335b0_ag_fw_name[fw_type]);
 			break;
 		case BCM4345_CHIP_ID:
 			if (chiprev == BCM43455C0_CHIP_REV)
