diff --git a/sound/soc/aml/m8/aml_i2s.c b/sound/soc/aml/m8/aml_i2s.c
index f48cd43..2ee9495 100755
--- a/sound/soc/aml/m8/aml_i2s.c
+++ b/sound/soc/aml/m8/aml_i2s.c
@@ -702,21 +702,23 @@ static int aml_i2s_copy_playback(struct snd_pcm_runtime *runtime, int channel,
     }
     if (access_ok(VERIFY_READ, buf, frames_to_bytes(runtime, count))) {
         if (runtime->format == SNDRV_PCM_FORMAT_S16_LE) {
-       int16_t * tfrom, *to, *left;
-       int16_t *right = NULL;
+            int16_t * tfrom, *to, *left;
+            int16_t *right = NULL;
 
-	tfrom = (int16_t *) ubuf;
+            tfrom = (int16_t *) ubuf;
 
-	for (j = 0; j < count; j++) {
-		to = (int16_t *) get_hw_buf_ptr(runtime, pos + j, align);
-		left = to;
-		if (runtime->channels != 1)
-			right = to + align;
+            for (j = 0; j < count; j++) {
+                to = (int16_t *) get_hw_buf_ptr(runtime, pos + j, align);
+                left = to;
+                if (runtime->channels != 1)
+                    right = to + align;
+
+                /* To avoid clipping on DIO2125/SGM8905 op-amp use -12 dB attenuation (2-bit shift) for PCM */
+                *left = IEC958_mode_codec ? (*tfrom++) : ((*tfrom++) >> 2);
+                if (runtime->channels != 1)
+                    *right = IEC958_mode_codec ? (*tfrom++) : ((*tfrom++) >> 2);
+            }
 
-		*left = (*tfrom++);
-		if (runtime->channels != 1)
-			*right = (*tfrom++);
-	}
         } else if (runtime->format == SNDRV_PCM_FORMAT_S24_LE && I2S_MODE == AIU_I2S_MODE_PCM24) {
             int32_t *tfrom, *to, *left, *right;
             tfrom = (int32_t*)ubuf;
@@ -730,8 +732,9 @@ static int aml_i2s_copy_playback(struct snd_pcm_runtime *runtime, int channel,
             }
             for (j = 0; j < n; j += 64) {
                 for (i = 0; i < 8; i++) {
-                    *left++  = (*tfrom ++);
-                    *right++  = (*tfrom ++);
+                    /* To avoid clipping on DIO2125/SGM8905 op-amp use -12 dB attenuation (2-bit shift) for PCM */
+                    *left++  = IEC958_mode_codec ? (*tfrom++) : ((*tfrom++) >> 2);
+                    *right++ = IEC958_mode_codec ? (*tfrom++) : ((*tfrom++) >> 2);
                 }
                 left += 8;
                 right += 8;
@@ -782,8 +785,9 @@ static int aml_i2s_copy_playback(struct snd_pcm_runtime *runtime, int channel,
             } else {
                 for (j = 0; j < n; j += 64) {
                     for (i = 0; i < 8; i++) {
-                        *left++  = (*tfrom ++) >> 8;
-                        *right++  = (*tfrom ++) >> 8;
+                        /* To avoid clipping on DIO2125/SGM8905 op-amp use -12 dB attenuation (2-bit shift) for PCM */
+                        *left++  = (IEC958_mode_codec ? (*tfrom++) : ((*tfrom++) >> 2)) >> 8;
+                        *right++ = (IEC958_mode_codec ? (*tfrom++) : ((*tfrom++) >> 2)) >> 8;
                     }
                     left += 8;
                     right += 8;
