diff --git a/drivers/amlogic/wifi/bcmdhd/Kconfig b/drivers/amlogic/wifi/bcmdhd/Kconfig
index 3b83c55..613abb7 100644
--- a/drivers/amlogic/wifi/bcmdhd/Kconfig
+++ b/drivers/amlogic/wifi/bcmdhd/Kconfig
@@ -18,6 +18,13 @@ config BCMDHD_NVRAM_PATH
 	---help---
 	  Path to the calibration file.
 
+config BCMDHD_CONFIG_PATH
+	depends on BCMDHD
+	string "Config path"
+	default "/etc/wifi/config.txt"
+	---help---
+	  Path to the driver configuration file.
+
 config BCMDHD_WEXT
 	bool "Enable WEXT support"
 	depends on BCMDHD && CFG80211 = n
