diff --git a/arch/arm/boot/dts/amlogic/meson8m2_wetek_core.dtd b/arch/arm/boot/dts/amlogic/meson8m2_wetek_core.dtd
index b4278c2..9265d1f 100644
--- a/arch/arm/boot/dts/amlogic/meson8m2_wetek_core.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_wetek_core.dtd
@@ -133,12 +133,12 @@ void root_func(){
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-		reserve-memory = <0x04800000  0x00100000>;
+		reserve-memory = <0x06000000  0x00100000>; /* fb0 size: 3840*2160*4*3 = 0x5EEC000 -> 96M (0x6000000) */
 		reserve-iomap = "true";
                 vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
 		4k2k_fb = <1>;
-		display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
+		display_size_default = <1920 1080 1920 3240 32>;
 	};
 
 //$$ DEVICE="deinterlace"
@@ -148,7 +148,7 @@ void root_func(){
 		compatible = "amlogic,deinterlace";
 		dev_name = "deinterlace";
 		status = "ok";
-		reserve-memory = <0x01800000>; //8x1920x1088x3/2=24M
+		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
 /// ***************************************************************************************
@@ -175,9 +175,22 @@ void root_func(){
                 compatible = "amlogic,vdec";
                 dev_name = "vdec.0";
                 status = "okay";
-                reserve-memory = <0x10000000>;
+                reserve-memory = <0x10000000>; // 256M
                 reserve-iomap = "true";
         };
+///	-	picture decoder
+//$$ MODULE="picture decoder"
+//$$ DEVICE="picdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 2 ="reg"
+    picdec{
+	compatible = "amlogic,picdec";
+	dev_name = "picdec";
+	status = "okay";
+                share-memory-name = "vdec0";
+                share-memory-offset = <0>;
+                share-memory-size = <0x04000000>; //64M
+	};
 
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
@@ -292,7 +305,7 @@ void root_func(){
 		dev_name = "mesonvout";
 		status = "okay";
 	};
-
+/*
 /// ***************************************************************************************
 ///	-	DISP&MM-A/V Amvideocap
 //$$ MODULE = "DISP&MM-Amvideocap"
@@ -306,7 +319,7 @@ void root_func(){
 		reserve-memory = <0x00600000>;//1920*1088*3
 		reserve-iomap = "true";
 	};
-
+*/
 /// ***************************************************************************************
 ///	-	RTC
 //$$ MODULE="RTC"
@@ -888,7 +901,7 @@ sdio{
 			lm-periph-id = <0>; /** lm name */
 			clock-src = "usb0"; /** clock src */
 			port-id = <0>;  /** ref to mach/usb.h */
-			port-type = <0>;	/** 0: otg, 1: host, 2: slave */
+			port-type = <1>;	/** 0: otg, 1: host, 2: slave */
 			port-speed = <0>; /** 0: default, high, 1: full */
 			port-config = <0>; /** 0: default */
 			port-dma = <0>; /** 0: default, 1: single, 2: incr, 3: incr4, 4: incr8, 5: incr16, 6: disable*/
