diff --git a/xbmc/guilib/GUIFadeLabelControl.cpp b/xbmc/guilib/GUIFadeLabelControl.cpp
index 51b087f..5d3be28 100644
--- a/xbmc/guilib/GUIFadeLabelControl.cpp
+++ b/xbmc/guilib/GUIFadeLabelControl.cpp
@@ -150,6 +150,7 @@ void CGUIFadeLabelControl::Process(unsigned int currentTime, CDirtyRegionList &d
     if (m_scroll)
     {
       m_textLayout.UpdateScrollinfo(m_scrollInfo);
+      MarkDirtyRegion();
     }
 
     CServiceBroker::GetWinSystem()->GetGfxContext().RemoveTransform();
