diff --git a/net/Makefile b/net/Makefile
index 72e64d5..67d460a 100644
--- a/net/Makefile
+++ b/net/Makefile
@@ -46,7 +46,7 @@ endif
 obj-$(CONFIG_IP_DCCP)		+= dccp/
 obj-$(CONFIG_IP_SCTP)		+= sctp/
 obj-$(CONFIG_RDS)		+= rds/
-#obj-$(CONFIG_WIRELESS)		+= wireless/
+obj-$(CONFIG_WIRELESS)		+= wireless/
 obj-$(CONFIG_MAC80211)		+= mac80211/
 obj-$(CONFIG_TIPC)		+= tipc/
 obj-$(CONFIG_NETLABEL)		+= netlabel/
