diff --git a/arch/arm/boot/dts/meson8b_odroidc.dts b/arch/arm/boot/dts/meson8b_odroidc.dts
index 52f3e5f..5147f0e 100644
--- a/arch/arm/boot/dts/meson8b_odroidc.dts
+++ b/arch/arm/boot/dts/meson8b_odroidc.dts
@@ -94,7 +94,7 @@
                 vmode = <0>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
                 scale_mode = <1>; /*0:default 1:new*/
                 4k2k_fb = <0>;
-                display_size_default = <1280 720 1280 1440 32>; //1920*1080*4*3 = 0x17BB000
+                display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
         };
 
         mesonstream{
@@ -474,6 +474,7 @@
         compatible = "amlogic,aml-spdif-codec";
         dev_name = "spdif-dit.0";
         pinctrl-names = "aml_audio_spdif";
+        pinctrl-0 = <&audio_spdif_pins>;
         status = "okay";
     };
 
@@ -515,14 +516,13 @@
                 status = "okay";
         };
 
-        meson-ir {
-                compatible = "amlogic,meson6-ir";
-                reg = <0xc8100580 0x20>;
-                interrupts = <0 15 1>;
-                pinctrl-names = "default";
-                pinctrl-0 = <&remote_pins>;
-                pulse-inverted;
+    meson-remote{
+                compatible = "amlogic,aml_remote";
+                dev_name = "meson-remote";
                 status = "ok";
+                ao_baseaddr = <0xf3100480>;
+                pinctrl-names="default";
+                pinctrl-0=<&remote_pins>;
         };
 
     spi@cc000000{
@@ -952,9 +952,10 @@
                 };
 
         audio_spdif_pins:audio_pin1{
-            amlogic,setmask=<10 0x8>;       /*spdif_out*/
-            amlogic,clrmask=<10 0x80002000>;    /*spdif_out*/
-            amlogic,pins ="GPIOAO_6";        /*spdif_out*/
+            amlogic,setmask=<10 0x00010000>;
+            amlogic,clrmask=<10 0x00040000
+                              1 0x00002000>;
+            amlogic,pins ="GPIOAO_6";
         };
 
         audio_pins:audio_pin{
