diff --git a/arch/arm/boot/dts/amlogic/meson8b_odroidc.dtd b/arch/arm/boot/dts/amlogic/meson8b_odroidc.dtd
index 3148810..a2a53d2 100755
--- a/arch/arm/boot/dts/amlogic/meson8b_odroidc.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_odroidc.dtd
@@ -641,7 +641,7 @@ void root_func(){
 //$$ L2 PROP_U32 16*3 = "dvfs_table"
         vcck_dvfs {
             dvfs_id     = <1>;                                  /** must be value of (1 << n)            */
-            table_count = <12>;                                 /** must be correct count for dvfs_table */
+            table_count = <13>;                                 /** must be correct count for dvfs_table */
             change-frequent-only;
             dvfs_table  = <
             /* NOTE: frequent in this table must be ascending order */
@@ -656,8 +656,9 @@ void root_func(){
                  816000         850000      850000
                 1008000         875000      875000
                 1200000         925000      925000
-                1320000        1000000     1000000
-                1488000        1075000     1075000
+                1320000        1100000     1100000
+                1488000        1100000     1100000
+                1536000        1140000     1140000
             >;
         };
     };
@@ -761,6 +762,7 @@ void root_func(){
         compatible = "amlogic,aml-spdif-codec";
         dev_name = "spdif-dit.0";
         pinctrl-names = "aml_audio_spdif";
+        pinctrl-0 = <&audio_spdif_pins>;
         status = "okay";
     };
 
@@ -780,45 +782,27 @@ void root_func(){
         status = "okay";
     };
 
-//$$ DEVICE="dummy_codec"
+//$$ DEVICE="pcm5102"
 //$$ L2 PROP_STR = "status"
-    dummy_codec{
-        compatible = "amlogic,aml_dummy_codec";
-        dev_name = "dummy_codec.0";
+    pcm5102{
+        compatible = "hardkernel,pcm5102";
+        dev_name = "pcm5102.0";
         status = "okay";
     };
 
-    aml_audio_codec{
-        compatible = "amlogic,audio_codec";
-        dummy:dummy{
-          codec_name = "dummy_codec";
-          status = "okay";
-        };
-    };
-
 //$$ DEVICE="aml_m8_sound_card"
-//$$ L2 PROP_STR = "aml,sound_card"
-//$$ L2 PROP_STR = "aml,codec_dai"
-//$$ L2 PROP_STR 6*2 = "aml,audio-routing"
-//$$ L2 PROP_STR = "mute_gpio"
-//$$ L2 PROP_U32 5 = "hp_paraments"
-//$$ L2 PROP_STR = "pinctrl-names"
-//$$ L2 PROP_CHOICE "Audio_pin_0_match" = "pinctrl-0"
 //$$ L2 PROP_STR = "status"
     aml_m8_sound_card{
         compatible = "sound_card, aml_snd_m8";
-        aml,sound_card = "AML-M8AUDIO";
-        aml,codec_dai = "AML-M8","dummy_codec";
-        aml,audio-routing-amlm8 =
-            "Ext Spk","LINEOUTL",
-            "Ext Spk","LINEOUTR",
-            "HP","HP_L",
-            "HP","HP_R";
-        aml,audio-routing-dummy=
-            "Ext Spk","LOUTL",
-            "Ext Spk","LOUTR";
+        aml,sound_card = "ODROID-HDMI";
         sleep_time = <130>;
-        pinctrl-names = "aml_snd_m8";
+        status = "okay";
+    };
+
+    odroid_sound_card{
+        compatible = "sound_card, odroid_snd";
+        aml,sound_card = "ODROID-DAC";
+        pinctrl-names = "odroid_i2s";
         pinctrl-0 = <&audio_pins>;
         status = "okay";
     };
@@ -987,10 +971,10 @@ void root_func(){
 //$$ L2 PROP_STR = "product_desc"
 //$$ L2 PROP_STR = "cec_osd_string"
         vend_data: vend_data{
-            vendor_name = "Amlogic";           /* Max Chars: 8     */
+            vendor_name = "Hardkernel";           /* Max Chars: 8     */
             vendor_id = <0x000000>;                 /* Refer to http://standards.ieee.org/develop/regauth/oui/oui.txt   */
-            product_desc = "M8 MBox SKTv1";        /* Max Chars: 16    */
-            cec_osd_string = "Amlogic MBox";        /* Max Chars: 14    */
+            product_desc = "ODROID-C1";        /* Max Chars: 16    */
+            cec_osd_string = "ODROID-C1";        /* Max Chars: 14    */
         };
 
 //$$ MATCH "HDMI_pwr-ctrl_match" = <&pwr_ctrl>
@@ -1434,9 +1418,10 @@ void root_func(){
 //$$ L2 PROP_U32 2 = "amlogic,clrmask"
 //$$ L2 PROP_STR 1 = "amlogic,pins"
         audio_spdif_pins:audio_pin1{
-            amlogic,setmask=<10 0x8>;       /*spdif_out*/
-            amlogic,clrmask=<10 0x80002000>;    /*spdif_out*/
-            amlogic,pins ="GPIOAO_6";        /*spdif_out*/
+            amlogic,setmask=<10 0x00010000>;
+            amlogic,clrmask=<10 0x00040000
+                              1 0x00002000>;
+            amlogic,pins ="GPIOAO_6";
         };
 
 //$$ MATCH "Audio_pin_0_match" = "&audio_pins"
@@ -1444,8 +1429,7 @@ void root_func(){
 //$$ L2 PROP_U32 10 = "amlogic,clrmask"
 //$$ L2 PROP_STR 6 = "amlogic,pins"
         audio_pins:audio_pin{
-            amlogic,setmask=<10 0x78000008>;
-            amlogic,clrmask=<10 0x80002000>;
+            amlogic,setmask=<10 0x78000000>;
             amlogic,pins = "GPIOAO_8","GPIOAO_9","GPIOAO_10","GPIOAO_11";
         };
 
@@ -1494,7 +1478,7 @@ void root_func(){
         compatible = "gpio-leds";
         /* Blue LED */
         hearbeat {
-            label = "blue:heartbeart";
+            label = "blue:heartbeat";
             gpios = "GPIOAO_13";
             default-state = "off";
             linux,default-trigger = "heartbeat";
