diff --git a/arch/arm/boot/dts/meson8b_odroidc.dts b/arch/arm/boot/dts/meson8b_odroidc.dts
index 5b519aa..d35ff55 100755
--- a/arch/arm/boot/dts/meson8b_odroidc.dts
+++ b/arch/arm/boot/dts/meson8b_odroidc.dts
@@ -521,19 +521,18 @@
         status = "okay";
     };
 
-        saradc{
+    saradc{
                 compatible = "amlogic,saradc";
                 status = "okay";
         };
 
-        meson-ir {
-                compatible = "amlogic,meson6-ir";
-                reg = <0xc8100580 0x20>;
-                interrupts = <0 15 1>;
-                pinctrl-names = "default";
-                pinctrl-0 = <&remote_pins>;
-                pulse-inverted;
+    meson-remote{
+                compatible = "amlogic,aml_remote";
+                dev_name = "meson-remote";
                 status = "ok";
+                ao_baseaddr = <0xf3100480>;
+                pinctrl-names="default";
+                pinctrl-0=<&remote_pins>;
         };
 
     spi@cc000000{
