diff --git a/addons/resource.language.en_gb/resources/strings.po b/addons/resource.language.en_gb/resources/strings.po
index 03b5894..298ccdf 100644
--- a/addons/resource.language.en_gb/resources/strings.po
+++ b/addons/resource.language.en_gb/resources/strings.po
@@ -19781,7 +19781,23 @@ msgctxt "#36436"
 msgid "Select OS keyboard layout."
 msgstr ""
 
-#empty strings from id 36437 to 36441
+#: system/settings/settings.xml
+msgctxt "#36437"
+msgid "Override power options"
+msgstr ""
+
+#: system/settings/settings.xml
+msgctxt "#36438"
+msgid "Force suspend instead of power off"
+msgstr ""
+
+#. Description for setting with label #36438 "Force suspend instead of power off"
+#: system/settings/settings.xml
+msgctxt "#36439"
+msgid "This option may be a solution for some TV boxes if turning on by remote control does not work after shutting down. ATTENTION! Use only in case of detailed condition, otherwise it may cause a malfunction!"
+msgstr ""
+
+#empty strings from id 36440 to 36441
 
 #. Description of setting "System -> Audio output -> Volume control steps" with label #1302
 #: system/settings/settings.xml
diff --git a/system/settings/settings.xml b/system/settings/settings.xml
index 19cfe2d..f19b7fc 100755
--- a/system/settings/settings.xml
+++ b/system/settings/settings.xml
@@ -2887,7 +2887,14 @@
           </control>
         </setting>
       </group>
-      <group id="2" label="14256">
+      <group id="2" label="36437">
+        <setting id="powermanagement.forcesuspend" type="boolean" label="36438" help="36439">
+          <level>2</level>
+          <default>false</default>
+          <control type="toggle" />
+        </setting>
+      </group>
+      <group id="3" label="14256">
         <setting id="powermanagement.wakeonaccess" type="boolean" label="13026" help="36350">
           <level>1</level>
           <default>false</default>
diff --git a/xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp b/xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp
index fec636a..722861f 100644
--- a/xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp
+++ b/xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp
@@ -9,6 +9,9 @@
 
 #include "LogindUPowerSyscall.h"
 #include "utils/log.h"
+#include "settings/SettingsComponent.h"
+#include "settings/Settings.h"
+#include "ServiceBroker.h"
 
 #include <string.h>
 #include <unistd.h>
@@ -76,7 +79,14 @@ CLogindUPowerSyscall::~CLogindUPowerSyscall()
 
 bool CLogindUPowerSyscall::Powerdown()
 {
-  return LogindSetPowerState("PowerOff");
+  if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(CSettings::SETTING_POWERMANAGEMENT_FORCESUSPEND)) {
+    CLog::Log(LOGDEBUG, "LogindUPowerSyscall - power state override: PowerOff -> Suspend");
+    return LogindSetPowerState("Suspend");
+  }
+  else
+  {
+    return LogindSetPowerState("PowerOff");
+  }
 }
 
 bool CLogindUPowerSyscall::Reboot()
diff --git a/xbmc/settings/Settings.cpp b/xbmc/settings/Settings.cpp
index 829e7b5..c4df38d 100644
--- a/xbmc/settings/Settings.cpp
+++ b/xbmc/settings/Settings.cpp
@@ -383,6 +383,7 @@ const std::string CSettings::SETTING_POWERMANAGEMENT_SHUTDOWNTIME = "powermanage
 const std::string CSettings::SETTING_POWERMANAGEMENT_SHUTDOWNSTATE = "powermanagement.shutdownstate";
 const std::string CSettings::SETTING_POWERMANAGEMENT_WAKEONACCESS = "powermanagement.wakeonaccess";
 const std::string CSettings::SETTING_POWERMANAGEMENT_WAITFORNETWORK = "powermanagement.waitfornetwork";
+const std::string CSettings::SETTING_POWERMANAGEMENT_FORCESUSPEND = "powermanagement.forcesuspend";
 const std::string CSettings::SETTING_DEBUG_SHOWLOGINFO = "debug.showloginfo";
 const std::string CSettings::SETTING_DEBUG_EXTRALOGGING = "debug.extralogging";
 const std::string CSettings::SETTING_DEBUG_SETEXTRALOGLEVEL = "debug.setextraloglevel";
diff --git a/xbmc/settings/Settings.h b/xbmc/settings/Settings.h
index d0743b3..64819ce 100644
--- a/xbmc/settings/Settings.h
+++ b/xbmc/settings/Settings.h
@@ -343,6 +343,7 @@ public:
   static const std::string SETTING_POWERMANAGEMENT_SHUTDOWNSTATE;
   static const std::string SETTING_POWERMANAGEMENT_WAKEONACCESS;
   static const std::string SETTING_POWERMANAGEMENT_WAITFORNETWORK;
+  static const std::string SETTING_POWERMANAGEMENT_FORCESUSPEND;
   static const std::string SETTING_DEBUG_SHOWLOGINFO;
   static const std::string SETTING_DEBUG_EXTRALOGGING;
   static const std::string SETTING_DEBUG_SETEXTRALOGLEVEL;
diff --git a/xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp b/xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp
index fec636a..408667d 100644
--- a/xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp
+++ b/xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp
@@ -36,10 +36,10 @@ CLogindUPowerSyscall::CLogindUPowerSyscall()
   if (!m_hasUPower)
     CLog::Log(LOGINFO, "LogindUPowerSyscall - UPower not found, battery information will not be available");
 
-  m_canPowerdown = LogindCheckCapability("CanPowerOff");
-  m_canReboot    = LogindCheckCapability("CanReboot");
-  m_canHibernate = LogindCheckCapability("CanHibernate");
-  m_canSuspend   = LogindCheckCapability("CanSuspend");
+  m_canPowerdown = true;
+  m_canReboot    = true;
+  m_canHibernate = false;
+  m_canSuspend   = false;
 
   InhibitDelayLockSleep();
 
